### PR TITLE
ecdsa: follow the DynAssociatedAlgorithmIdentifier changes

### PR DESCRIPTION
**Pull request by lumag**
Return Result from algorithm_identifier().

Depends on https://github.com/RustCrypto/formats/pull/974

### CODE REVIEW

**Comment by lumag** [comment]
I can not reproduce the build failure locally. Probably a git revision got cached somewhere and is not updated.

**Comment by tarcieri** [comment]
You need to run `cargo update -p spki`. I would not have expected the build to work locally without doing this.

**Comment by tarcieri** [comment]
Is it possible you're not checking in `Cargo.lock`?

You didn't update it in this PR, so I would expect it to fail because you're using an out-of-date version

File: ecdsa/src/lib.rs [hunk 1]
```diff
@@ -672,11 +674,11 @@ impl<C> DynAssociatedAlgorithmIdentifier for SignatureWithOid<C>
 where
     C: PrimeCurve,
 {
-    fn algorithm_identifier(&self) -> AlgorithmIdentifierOwned {
-        AlgorithmIdentifierOwned {
+    fn algorithm_identifier(&self) -> SpkiResult<AlgorithmIdentifierOwned> {
```
Comment (tarcieri) [inline ecdsa/src/lib.rs:677]:
```suggestion
    fn algorithm_identifier(&self) -> spki::Result<AlgorithmIdentifierOwned> {
```

Comment (lumag) [reply ecdsa/src/lib.rs:677]: @tarcieri there is no direct dependency on `spki`. Would you prefer:
- add spki as a dep?
- use whatever..spki to make `spki` available as a name
- use full path including elliptic-curves::pkcs8:...

File: ecdsa/src/lib.rs [hunk 2]
```diff
@@ -117,7 +117,9 @@ use elliptic_curve::pkcs8::spki::{
 use serdect::serde::{de, ser, Deserialize, Serialize};
 
 #[cfg(all(feature = "alloc", feature = "pkcs8"))]
-use elliptic_curve::pkcs8::spki::{AlgorithmIdentifierOwned, DynAssociatedAlgorithmIdentifier};
+use elliptic_curve::pkcs8::spki::{
+    AlgorithmIdentifierOwned, DynAssociatedAlgorithmIdentifier, Result as SpkiResult,
```
Comment (tarcieri) [inline ecdsa/src/lib.rs:121]:
```suggestion
    self, AlgorithmIdentifierOwned, DynAssociatedAlgorithmIdentifier,
```

File: ecdsa/src/lib.rs (hunk 1 above)
Comment (tarcieri) [reply ecdsa/src/lib.rs:677]: See: https://github.com/RustCrypto/signatures/pull/695/files#r1157862631

File: ecdsa/src/lib.rs (hunk 2 above)
Comment (lumag) [reply ecdsa/src/lib.rs:121]: ack,

Comment (lumag) [reply ecdsa/src/lib.rs]: done

**Review by tarcieri — APPROVED** [review]

**Pull request merged**